### PR TITLE
Fixed #26441 -- Added model Field.db_check() method.

### PR DESCRIPTION
**Pull request by MarkusH**
Working on a separate patch I had to get the constraints w/o triggering getting the field type.

### CODE REVIEW

**Comment by timgraham** [comment]
Could you please add a docstring for the method? Should it be documented in docs/howto/custom-model-fields.txt as well? How about "Added model Field.db_check() method." for the commit message?

**Comment by MarkusH** [comment]
Talking to @mjtamlyn we certainly should make `db_type()`, `db_check()` and `db_parameters()` public API to ensure at least `db_parameters()` is a single valid entry point to get all information.

Since only the `db_type()` is documented so far and there's no way to override the check constrains except for overriding the private `db_parameters()` (which will still work) I'd not make it a public API just yet.

File: django/db/models/fields/__init__.py [hunk 1]
```diff
@@ -594,6 +594,18 @@ def clean(self, value, model_instance):
         self.run_validators(value)
         return value
 
+    def db_check(self, connection):
+        """
+        Returns the database column constraints for this field, for the
```
Comment (timgraham) [inline django/db/models/fields/__init__.py]: "Return" (PEP257 style) for new docstrings.

**Comment by timgraham** [comment]
Looks okay. I don't find the name very intuitive but it could just be unfamiliarity in this area.

**Comment by MarkusH** [comment]
Merged in 103d4e1d65b8e198a237ef65e709d15459d709cb with `db_check()` after another naming round with @mjtamlyn . The method returns the `SQL CHECK` constraint for a field, not all constraints.

**Pull request closed without merging**